junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/ScriptExecutionCondition.java

```diff
@@ -111,66 +91,23 @@ private String createSource(String[] lines) {
 		return String.join(System.lineSeparator(), lines);
 	}
 
-	private ScriptExecutionManager getScriptExecutionManager(ExtensionContext context) {
-		return context.getRoot().getStore(NAMESPACE).getOrComputeIfAbsent(ScriptExecutionManager.class);
-	}
-
-	private Bindings createBindings(ExtensionContext context) {
-		ScriptAccessor configurationParameterAccessor = new ScriptAccessor.ConfigurationParameterAccessor(context);
-		Bindings bindings = new SimpleBindings();
-		bindings.put(Script.BIND_JUNIT_TAGS, context.getTags());
-		bindings.put(Script.BIND_JUNIT_UNIQUE_ID, context.getUniqueId());
-		bindings.put(Script.BIND_JUNIT_DISPLAY_NAME, context.getDisplayName());
-		bindings.put(Script.BIND_JUNIT_CONFIGURATION_PARAMETER, configurationParameterAccessor);
-		return bindings;
+	private Worker getScriptExecutionWorker(ExtensionContext context) {
+		return context.getRoot().getStore(NAMESPACE).getOrComputeIfAbsent(Worker.class, this::createWorker,
+			Worker.class);
 	}
 
-	ConditionEvaluationResult evaluate(ScriptExecutionManager manager, Script script, Bindings bindings) {
-		if (script == null) {
-			return null;
-		}
+	private Worker createWorker(Class<Worker> key) {
+		// Create worker via reflection to hide the `java.scripting` dependencies.
 		try {
-			Object result = manager.evaluate(script, bindings);
-			return computeConditionEvaluationResult(script, result);
+			return (Worker) Class.forName(
+				"org.junit.jupiter.engine.extension.ScriptExecutionWorker").getDeclaredConstructor().newInstance();
 		}
-		catch (ScriptException e) {
-			throw new ScriptEvaluationException("Script evaluation failed for: " + script.getAnnotationAsString(), e);
+		catch (ReflectiveOperationException e) {
```

Comment:
I think we should make this more explicit, i.e. check whether a class from `javax.script` is present. Moreover, we should provide a hint to the user, possibly a link to the User Guide, of how to fix the setup.

Reply:
> WORK-IN-PROGRESS

:)

Reply:
Well, if you don't want "helpful suggestions", don't open a PR... 😜

Reply:
*+1* 😝

Reply:
🤓🤓